lib/spack/spack/database.py

```diff
@@ -1112,6 +1117,7 @@ def _add(
             # the original hash of concrete specs.
             new_spec._mark_concrete()
             new_spec._hash = key
+            new_spec._full_hash = spec._full_hash
```

Comment:
Should this be conditional, similar to `build_hash`, just below?  Sometimes we can get a concrete spec where the `full_hash` is `None`, we get that in the yaml.

Reply:
Ah jeez, my bad.  That comment was intended for another section of the code, in `spec.py` [here](https://github.com/spack/spack/pull/19209/files#diff-9e28cdd6f9cd66e61101de6e9d22b1c33561959e4305c63147bbfbf7b22da936R1645).  Ugh, sorry for the confusion.